src/python/pants/core/goals/fmt.py

```diff
@@ -3,132 +3,30 @@
-        def prep_output(s: bytes) -> str:
-            return strip_v2_chroot_path(s) if strip_chroot_path else s.decode()
-
-        return FmtResult(
-            input=request.snapshot,
-            output=await Get(Snapshot, Digest, process_result.output_digest),
-            stdout=prep_output(process_result.stdout),
-            stderr=prep_output(process_result.stderr),
-            formatter_name=request.tool_name,
-        )
-
-    def __post_init__(self):
-        # NB: We debug log stdout/stderr because `message` doesn't log it.
-        log = f"Output from {self.formatter_name}"
-        if self.stdout:
-            log += f"\n{self.stdout}"
-        if self.stderr:
-            log += f"\n{self.stderr}"
-        logger.debug(log)
-
-    @property
-    def did_change(self) -> bool:
-        return self.output != self.input
-
-    def level(self) -> LogLevel | None:
-        return LogLevel.WARN if self.did_change else LogLevel.INFO
-
-    def message(self) -> str | None:
-        message = "made changes." if self.did_change else "made no changes."
-
-        # NB: Instead of printing out `stdout` and `stderr`, we just print a list of files which
-        # were changed/added/removed. We do this for two reasons:
-        #   1. This is run as part of both `fmt` and `lint`, and we want consistent output between both
-        #   2. Different formatters have different stdout/stderr. This way is consistent across all
-        #       formatters.
-        # We also allow added/removed files because the `fmt` goal is more like a `fix` goal.
-        # (See https://github.com/pantsbuild/pants/issues/13504)
-        if self.did_change:
-            snapshot_diff = SnapshotDiff.from_snapshots(self.input, self.output)
-            output = "".join(
-                f"\n  {file}"
-                for file in itertools.chain(
-                    snapshot_diff.changed_files,
-                    snapshot_diff.their_unique_files,  # added files
-                    snapshot_diff.our_unique_files,  # removed files
-                    # NB: there is no rename detection, so a renames will list
-                    # both the old filename (removed) and the new filename (added).
-                )
-            )
-        else:
-            output = ""
-
-        return f"{self.formatter_name} {message}{output}"
-
-    def cacheable(self) -> bool:
-        """Is marked uncacheable to ensure that it always renders."""
-        return False
-
-
-Partitions = UntypedPartitions[PartitionKeyT, str]
+FmtResult = FixResult
 
 
 @union
-class FmtRequest(LintRequest):
+class FmtRequest(FixRequest):
     is_formatter = True
-
-    @distinct_union_type_per_subclass(in_scope_types=[EnvironmentName])
-    @frozen_after_init
-    @dataclass(unsafe_hash=True)
-    class SubPartition(LintRequest.SubPartition):
-        snapshot: Snapshot
-
-        @property
-        def files(self) -> tuple[str, ...]:
-            return self.elements
+    is_fixer = False
```

Comment:
(this may be way late, as I know this topic has been ongoing for a long time now) I don't mind the `fix` as a goal name, but I start squinting slightly when I see `is_fixer` as a flag for a FixRequest type.. To me I'd prefer to reason that the term "fix" or "fixer" would be the umbrella term for the whole goal and everything it does. Having a flag to say something isn't a fixer then makes no sense to me.
Maybe we could consider using another term for this on the implementation side of things closer to what they actually do rather than what they are? (not feeling I'm able to phrase this properly).

How about: `is_formatter => fixes_style` and `is_fixer => fixes_semantics` as examples for what I'm looking for. Not saying those are the best either but side steps my issue I was going for  in the previous paragraph. Feel free to disagree, though :)

Reply:
Or is this related to the fact that `fix` also incorporates `fmt`? Then maybe the interface for the `FixRequest` is what I'm squinting at.. ?

Reply:
:O